pkg/storage/replica.go

```diff
@@ -905,6 +905,10 @@ func (r *Replica) updateProposalQuotaRaftMuLocked(
 	// ourselves via looking at MsgAppResp messages would be overkill. Perhaps
 	// another accessor on RawNode.
 	status := r.raftStatusRLocked()
+	if status == nil {
```

Comment:
it should be fine here, if we get to this codepath if you look we're not actually doing anything above. The mistake here was assuming proper initialization but as is in the case of the failures above, does not happen if the replica is destroyed ([here](https://github.com/irfansharif/cockroach/blob/79d6e2207fcb7c2977591ae856a6b0fbcda6a0d3/pkg/storage/replica.go#L2990-L2992), then [here](https://github.com/irfansharif/cockroach/blob/79d6e2207fcb7c2977591ae856a6b0fbcda6a0d3/pkg/storage/replica.go#L457) never reaching [here](https://github.com/irfansharif/cockroach/blob/79d6e2207fcb7c2977591ae856a6b0fbcda6a0d3/pkg/storage/replica.go#L486)).